### PR TITLE
Integrating product type with be

### PR DESCRIPTION
**Pull request by derpdead**
# Description

Issue # (#418 #384)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Unit Tests
- [x] e2e Test

# Checklist:

- [x] I have read the contribution requirements and fulfil them.
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: modules/@ergo/collections/components/Modals/AddProductsBySKUModalForm.vue [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+            this.$emit('close');
+        },
+        onAdd() {
+            this.removeValidationErrors();
+            const preValidationErrors = {};
+            const data = {
+                skus: this.productSkus,
+            };
+
+            if (!isEmpty(preValidationErrors)) {
```
Comment (bleto) [inline modules/@ergo/collections/components/Modals/AddProductsBySKUModalForm.vue:75]: this preValidation is empty, it is ok ?

File: modules/@ergo/products/components/Form/ProductForm.vue [hunk 2]
```diff
@@ -8,6 +8,18 @@
...
+                    :value="type"
+                    solid
+                    regular
+                    required
+                    :disabled="isDisabled"
+                    label="Product type"
+                    :options="productTypesValues"
+                    @input="setProductType" />
+                <ProductAttributesBindingFormSection
+                    v-show="productTypeKey === 'VARIABLE-PRODUCT'"
```
Comment (bleto) [inline modules/@ergo/products/components/Form/ProductForm.vue:21]: move `VARIABLE-PRODUCT` to deafults or const valiable.

File: modules/@ergo/products/components/Form/Section/ProductAttributesBindingFormSection.vue [hunk 3]
```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
+ * See LICENSE for license details.
+ */
+<template>
+    <FormListSection
+        :disabled="disabled || bindingAttributesIds.length === selectAttributes.length"
```
Comment (bleto) [inline modules/@ergo/products/components/Form/Section/ProductAttributesBindingFormSection.vue:7]: maybe move this contition to computed property.

File: modules/@ergo/products/components/Pages/ProductPage.vue [hunk 4]
```diff
@@ -87,11 +87,30 @@ export default {
...
+                this.$hasAccess,
+                this.$router.options.routes,
+                this.$route,
+            );
+
+            switch (getKeyByValue(this.productTypes, this.type)) {
+            case 'VARIABLE-PRODUCT':
+                return tabs.filter(tab => tab.title !== 'Group');
+            case 'GROUPING-PRODUCT':
+                return tabs.filter(tab => tab.title !== 'Variants');
```
Comment (bleto) [inline modules/@ergo/products/components/Pages/ProductPage.vue:110]: 'VARIABLE-PRODUCT' &  'GROUPING-PRODUCT' to defaults

File: modules/@ergo/collections/components/Modals/AddProductsBySKUModalForm.vue (hunk 1 above)
Comment (derpdead) [reply modules/@ergo/collections/components/Modals/AddProductsBySKUModalForm.vue:75]: Yes, cuz it has no errors

**Pull request merged**